proposals/NNNN-keep-rc-app-hmi-level-when-disable-rc.md

```diff
@@ -0,0 +1,48 @@
+# SDL shall not put RC applications to HMI level NONE when user disables RC in HMI
+
+* Proposal: [SDL-NNNN](NNNN-keep-rc-app-hmi-level-when-disable-rc.md)
+* Author: [Zhimin Yang](https://github.com/smartdevicelink/yang1070)
+* Status: **Awaiting review**
+* Impacted Platforms: [ Core / Web]
+
+## Introduction
+
+SDL shall not put `REMOTE_CONTROL` `appHMIType` applications to HMI level `NONE` when user disables remote control in HMI.
+
+## Motivation
+The asked behavior is in the same line with other existing features, like the permission of function groups. If a user disables the application's permission to access vehicle signals like gps, speed, etc. the application will keep its current HMI level and will not be put into HMI level `NONE` by SDL. Furthermore, SDl allows activating  an application using such signals. However, SDL rejects any RPC that relates to the disallowed signals. In this proposal, we propose the following changes. SDL shall keep a RC application's HMI status unchanged if the user disables RC. SDL can activate a RC application even if RC is disabled. 
+
+
+## Proposed solution
+
+We propose the changes as in the following tables:
+
+  | Current | Expected SDL core change
+-- | -- | --
+1 | HMI sends a notification to SDL that RC is disabled | No change
+2 | SDL puts RC apps from `FULL`/`LIMITED` to `NONE` | SDL does not change apps' HMI level
+3 | SDL sends `OnHMIStatus` (`NONE`) to all RC applications that put to `NONE` | SDL does not send `OnHMIStatus`
+4 | SDL keeps RC apps registered | No change
+5 | SDL unsubscribes all RC apps’ RC modules from HMI | No change
+6 | SDL releases all RC modules allocated to RC apps and sends `OnRCStatus` notifications | No change
+7 | SDL ignores `OnInteriorVehicleData` from HMI and SDL does not send `OnInteriorVehicleData` to apps | No change
+8 | SDL disallows RC related RPC, such as `GetInteriorVehicleData`, `SetInteriorVehicleData` and `ButtonPress` requests from apps | No change
+9 | HMILevel cannot change to other than `NONE` until RC functionality is enabled | HMI level can change like regular apps. In addtion, A RC app can be activated.
```

Comment:
Please update to "...cannot change to anything other than `NONE` until..." and fix typo "In addition, a RC app..."

Reply:
updated